Revise tree & panel container sizing / layout

rlskoeser opened the pull request:
- make tree and panel proportional on desktop, as decided at today's co-working session
- resolve problem with leaf details height causing page to overflow/jump
- improve mobile layout/sizing
- adjust placement of active tag when no leaf is open; addresses #161
- make panel close button on mobile _not_ full width (as decided at yesterday's meeting)
- add gradient at top and bottom of panel so scrolling content will fade out (bottom disabled on desktop due to a vertical position/size issue I still need to resolve)
- rearranged content order in the dom for accessibility, and then position as before using `grid-area`

Notes:
-  I made a guess for how large the tree container should be on mobile. I can adjust when you decide what it should be.
- I've disabled scrolling on the body on mobile - will revisit when I implement the footer. It needs to have scrolling disabled when the panel is visible, otherwise you can scroll further down than we want. 

revisions:
- [x] fix active tag placement with no leaf on desktop
- [x] switch to rems for the panel min/max width
- [x] bump tablet breakpoint from 768 to 790
- [x] make height of tree on mobile a scss variable

gissoo commented:
Thanks, @rlskoeser !! 

here are the issues I'm noticing:
- the jump is not happening anymore, but it seems like the intro panel's left and right edges are being slightly cut off / the shadow is missing 
<img width="542" alt="Screenshot 2023-04-25 at 8 53 17 PM" src="https://user-images.githubusercontent.com/44615389/234438631-1b601608-9a46-4779-8cab-bdd8578b3499.png">
<img width="483" alt="Screenshot 2023-04-25 at 8 53 30 PM" src="https://user-images.githubusercontent.com/44615389/234438654-3151b6b2-1448-4b3c-85e2-bd46086ad529.png">

- this white spot is appearing on safari and chrome, and it closes the leaf panel, it's part of the x button I guess
<img width="457" alt="Screenshot 2023-04-25 at 8 54 21 PM" src="https://user-images.githubusercontent.com/44615389/234438767-99c098d0-870f-4679-a5a6-2d4d52bef0b5.png">

- the placement of the active tag on mobile is fine, but on desktop it's weird - this is happening on all browsers right now
<img width="1680" alt="Screenshot 2023-04-25 at 8 50 34 PM" src="https://user-images.githubusercontent.com/44615389/234438350-991f8606-d5d2-4258-8b1a-4fb1d0846ead.png">

- the panel close button and the fade look great!! 
- regarding the tree size on mobile, I commented on our co-working session doc earlier, sorry, I think it would be good if we can see the entire tree (all the leaves and the trunk) by default while the intro panel. What do you think about that? 
Right now this is what I see on chrome and firefox mobile - there is still a lot of empty space on the panel – do y
![Screenshot_20230425_204447_Chrome](https://user-images.githubusercontent.com/44615389/234439420-ccf339b9-2fd5-43d6-8f12-6b220d9a2dfb.jpg)
![Screenshot_20230425_204444_Firefox](https://user-images.githubusercontent.com/44615389/234439426-ee24dd8b-010f-4348-9ed9-c717ec99a688.jpg)

- regarding the panel resizing, I love it, it's really close! What is this width? I also commented on the co-working session doc that I think the min width should be 320px 
<img width="775" alt="Screenshot 2023-04-25 at 9 02 08 PM" src="https://user-images.githubusercontent.com/44615389/234439596-6fb9115b-0fba-4cb9-af6a-f5c4654f52c2.png">

rlskoeser commented:
@gissoo thank you for testing and finding all of these!

The sizing I'm using right now on desktop is 32% of the width is dedicated to the panel, the rest is the tree, with a column gap of 38px.  LMK if you want to try tweaking that percentage.

- I ran into the margin issue for the drop shadow on the panel before but forgot; I've added back margins and a comment why we need them
- fixed the weird white spot on the close button (it was an overflow problem but IDK why 🤷‍♀️ )
- added a min width to the panel; I think if the min width is larger than the % allotted in the grid, it will extend the page slightly so there is a horizontal scroll, but that seems ok and better to have the content readable (note: I should probably convert the pixel sizes to rem so they are proportional to the text)

hahahaha on the active tag with no leaf - it's getting the mobile styles! I guess I forgot to override what I changed to adjust the position on mobile for that case; will adjust.


Your sentence about mobile size of the tree got cut off, I'm not sure what you intend. You want the whole tree and trunk visible when the intro is showing? I think it will be too small and will waste vertical space, unless we resize when the intro isn't visible (which I'd prefer not to do). But maybe I'm misunderstanding. (Wish we'd been able to get to this part in our co-working session, would probably be easier to talk through and adjust at together.)

gissoo commented:
> @gissoo thank you for testing and finding all of these!
> 
> The sizing I'm using right now on desktop is 32% of the width is dedicated to the panel, the rest is the tree, with a column gap of 38px. LMK if you want to try tweaking that percentage.
> 

- @rlskoeser I think it looks fine! 

>     * I ran into the margin issue for the drop shadow on the panel before but forgot; I've added back margins and a comment why we need them
> 
>     * fixed the weird white spot on the close button (it was an overflow problem but IDK why 🤷‍♀️ )
> 

- thank you!

>     * added a min width to the panel; I think if the min width is larger than the % allotted in the grid, it will extend the page slightly so there is a horizontal scroll, but that seems ok and better to have the content readable (note: I should probably convert the pixel sizes to rem so they are proportional to the text)
> 

- converting the pixel sizes to rem makes sense – I didn't realize about the min width ...I can't test that right now to see the horizontal scroll. 

> 
> hahahaha on the active tag with no leaf - it's getting the mobile styles! I guess I forgot to override what I changed to adjust the position on mobile for that case; will adjust.
> 

:D 


> Your sentence about mobile size of the tree got cut off, I'm not sure what you intend. You want the whole tree and trunk visible when the intro is showing? I think it will be too small and will waste vertical space, unless we resize when the intro isn't visible (which I'd prefer not to do). But maybe I'm misunderstanding. (Wish we'd been able to get to this part in our co-working session, would probably be easier to talk through and adjust at together.)
>

- you're right, that will be too small. I also don't like the conditional resizing. What you have looks fine, we could make it smaller if we see that it's not enough (once we implement the new site title). I think it will be good enough though.
